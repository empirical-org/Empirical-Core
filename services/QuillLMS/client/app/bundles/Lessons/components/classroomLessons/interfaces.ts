import * as CLIntF from '../../interfaces/classroomLessons';

export type ClassroomSessionId = string

export type ClassroomUnitId = string

export interface ClassroomLessonSession {
  students: Students;
  presence: Presence;
  current_slide: string;
  modes: Modes;
  submissions: Submissions;
  selected_submissions: SelectedSubmissions;
  selected_submission_order: SelectedSubmissionOrder;
  timestamps: Timestamps;
  flaggedStudents: FlaggedStudents;
  public: Boolean | null;
  models: Models;
  prompts: Prompts;
  followUpActivityName: string | null;
}

export interface ClassroomLessonSessions {
 [sessionUID:string]: ClassroomLessonSession
}

export interface Students {
 [key:string]: string
}

export interface TeacherAndClassroomName {
  teacher: string,
  classroom: string
}

export interface Presence {
 [key:string]: boolean
}

export interface Modes {
 [key:string]: string
}

export interface Submissions {
 [key:string]: QuestionSubmissionsList
}

export interface QuestionSubmissionsList {
  [key:string]: QuestionSubmission
}

export interface QuestionSubmission {
  timestamp: string;
  data: any
}

export interface SelectedSubmissions {
  [key:string]: SelectedSubmissionsForQuestion
}

export interface SelectedSubmissionsForQuestion {
  [key:string]: boolean
}

export interface Timestamps {
  [key:string]: string
}

export interface FlaggedStudents {
  [key:string]: boolean
}

export interface Models {
  [key: string]: string
}

export interface Prompts {
  [key: string]: string
}

export interface ClassroomLessons {
  [key: string]: ClassroomLesson
}

export interface ClassroomLesson {
  title: string;
  lesson?: string|number;
  topic?: string;
  unit?: string;
  questions?: Questions;
  id?: string;
}

export interface Questions {
  [key: string]: Question
}

export interface Question {
  type: string;
  data: CLIntF.QuestionData;
}

export interface SelectedSubmissionOrder {
  [key:number]: Array<string>
}
