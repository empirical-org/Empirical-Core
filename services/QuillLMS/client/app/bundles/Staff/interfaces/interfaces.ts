export interface Concept {
  id?:string;
  name?:string;
  value?: string;
  label?: string;
  updatedAt?: string;
  visible?: Boolean;
  uid?: string;
  parent?:Concept;
  replacement?: Concept;
  description?: string;
  explanation?: string;
  changeLogs?: Array<ChangeLog>;
}

export interface ChangeLog {
  action: string;
  explanation: string;
  id?: string;
  conceptID?: string;
  user?: { name?: string };
  previousValue?: string;
  concept?: Concept
  createdAt?: number;
}

export interface LockerContentsInterface {
  label: string,
  route?: string,
  href?: string,
  emoji: string,
  emojiLabel: string,
  overrideTitleCase?: boolean,
  tooltipInfo?: string,
  lockers?: {
    [key:string]: string[]
  }
}

export interface LockerItemsInterface {
  [key:string]: LockerContentsInterface
}
