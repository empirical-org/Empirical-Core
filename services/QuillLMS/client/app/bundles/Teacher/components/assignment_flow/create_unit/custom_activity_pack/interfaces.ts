export interface Activity {
  name: string
  description: string
  flags: string[]
  id: number
  uid: string
  anonymous_path: string
  activity_classification: ActivityClassification
  activity_category: ActivityCategory
  activity_category_name: string
  activity_category_id: number
  standard_level: StandardLevel
  standard_level_id: number
  standard_level_name: string
  content_partners: ContentPartner[]
  topics: Topic[]
  readability_grade_level?: string,
  maximum_grade_level?: number,
  minimum_grade_level?: number
}

export interface ActivityClassification {
  alias: string
  description: string
  key: string
  id: number
}

export interface ActivityCategory {
  id: number
  name: string
  order_number?: number
}

interface StandardLevel {
  id: number
  name: string
}

export interface ContentPartner {
  id: number
  name: string
  description: string
}

export interface Topic {
  id: number
  name: string
  level: number
  parent_id?: number
}

export interface ActivityCategoryEditor {
  activityCategories: ActivityCategory[]
  getActivityCategories: () => void
  setActivityCategories: (activityCategories: ActivityCategory[]) => void
  selectedActivityCategoryId: number
  handleActivityCategorySelect: (activityCategoryId: number) => void
  timesSubmitted: number
}
