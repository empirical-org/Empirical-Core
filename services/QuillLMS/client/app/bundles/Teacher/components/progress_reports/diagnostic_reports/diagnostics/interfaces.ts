export interface Activity {
  activity_id: number,
  activity_name: string,
  assigned_count: number,
  assigned_date: string,
  classroom_id: number,
  classroom_name: string,
  classroom_unit_id: number,
  completed_count: number,
  post_test_id: number|null,
  skills_count: number,
  unit_id: number,
  unit_name: string,
}

export interface Diagnostic {
  name: string,
  pre: Activity,
  post?: Activity
}

export interface Classroom {
  name: string;
  id: string;
  diagnostics: Array<Diagnostic>;
}

export interface ConceptResults {
  pre?: ConceptResults;
  post?: ConceptResults;
  questions?: Question[]
}

interface Question {
  directions: string;
  prompt: string;
  answer: string;
  concepts: ConceptResult;
  question_number: string;
}

interface ConceptResult {
  id: number;
  name: string;
  correct: boolean;
  lastFeedback: string;
  attempt: number;
  answer: string;
  directions: string;
}

export interface SkillResults {
  skills: Array<Skill>
}

export interface SkillGroup {
  skills: Array<Skill>;
  acquired_skill_ids: number[];
  id: number;
  number_of_correct_skills_text: string;
  post_test_proficiency: string;
  pre_test_proficiency: string;
  proficiency_text: string;
  skill_group: string;
}

export interface Skill {
  pre?: Skill;
  post?: Skill;
  skill?: string;
  number_correct: number;
  number_incorrect: number;
  summary: string;
}

export interface SkillGroupSummary {
  name: string;
  description?: string;
  not_yet_proficient_in_post_test_student_names?: string[];
  not_yet_proficient_in_pre_test_student_names?: string[];
}

export interface StudentResult {
  name: string;
  id?: number;
  skill_groups?: SkillGroup[];
  total_acquired_skills_count?: number;
}

export interface OpenPopover {
  studentId?: number;
  skillGroupId?: number;
}

export interface Recommendation {
  activity_pack_id: number;
  name: string;
  activity_count: number;
  students: number[];
}

export interface LessonRecommendation {
  activities: LessonsActivity[];
  activity_pack_id: number;
  name: string;
  percentage_needing_instruction: number;
  previously_assigned: boolean;
  students_needing_instruction: string[];
}

export interface LessonsActivity {
  name: string;
  url: string;
}

export interface Student {
  id: number;
  name: string;
  completed: boolean;
}
