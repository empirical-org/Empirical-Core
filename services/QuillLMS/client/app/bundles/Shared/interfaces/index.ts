export { UserIdsForEvent, NumberFilterInputProps } from './misc'
export { Question, QuestionObject } from './question'
