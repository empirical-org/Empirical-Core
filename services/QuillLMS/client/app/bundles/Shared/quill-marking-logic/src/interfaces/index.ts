export interface Response {
  author?: string|null,
  feedback?: string|null,
  first_attempt_count?: number|null,
  child_count?: number|null,
  concept_results?: Array<ConceptResult>|null,
  conceptResults?: Array<ConceptResult>|null,
  count: number,
  created_at?: string,
  id?: number,
  key?: string,
  misspelled_words?: Array<string>;
  optimal?: Boolean|null,
  parent_uid?: string|null,
  parent_id?: number|string|null,
  parentId?: number|string|null,
  question_uid: string,
  sortOrder?: number,
  statusCode?: number
  uid?: string|null,
  updated_at?: string,
  text: string,
  weak?: Boolean|null,
  spelling_error?: Boolean|null,
}

export interface PartialResponse {
  author?: string|null,
  feedback?: string|null,
  first_attempt_count?: number|null,
  child_count?: number|null,
  concept_results?: Array<ConceptResult>|null,
  conceptResults?: Array<ConceptResult>|null,
  created_at?: string,
  id?: number,
  key?: string,
  misspelled_words?: Array<string>;
  optimal?: Boolean|null,
  parent_uid?: string|null,
  parent_id?: number|string|null,
  parentId?: number|string|null,
  sortOrder?: number,
  statusCode?: number
  uid?: string|null,
  updated_at?: string,
  weak?: Boolean|null,
  errorType?: string,
}


export interface ConceptResult {
 correct: Boolean,
 conceptUID: string,
 name?: string,
 key?: string
}

export interface FocusPoint {
  text: string,
  feedback: string,
  concept_uid?: string,
  concept_results?: Array<ConceptResult>,
  conceptResults?: Array<ConceptResult>,
  key?: string,
  name?: string
}

export interface IncorrectSequence {
  text: string,
  feedback: string,
  concept_results?: Array<ConceptResult>,
  conceptResults?: Array<ConceptResult>,
  caseInsensitive?: boolean|null,
  name?: string
}

export interface FeedbackObject {
  feedback: string
}

export interface GradingObject {
  response: string,
  responses: Array<Response>,
  focusPoints?: Array<FocusPoint>,
  incorrectSequences?: Array<IncorrectSequence>,
  spellCorrectedResponse?: string
}

export interface WordCountChange {
  min?: number,
  max?: number
}
