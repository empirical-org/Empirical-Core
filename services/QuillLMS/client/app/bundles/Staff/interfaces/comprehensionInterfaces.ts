export interface ActivityRouteProps {
  activityId: string,
  type?: string,
  promptConjunction?: string
}

export interface ActivityInterface {
  id?: string,
  parent_activity_id?: string,
  title: string,
  notes: string,
  // flag: string,
  scored_level: string,
  target_level: number,
  passages?: PassagesInterface[],
  prompts?: PromptInterface[],
  passage_attributes?: PassagesInterface[],
  prompt_attributes?: PromptInterface[]
}

export interface PromptInterface {
  id?: number,
  conjunction: string,
  text: string,
  max_attempts: number,
  max_attempts_feedback: string
}

export interface RegexRuleInterface {
  id?: number,
  rule_id?: number,
  regex_text: string,
  case_sensitive: boolean,
  sequence_type: string
}

export interface FlagInterface {
  label: string,
  value: {}
}

export interface ActivityRuleSetPrompt {
  id?: number,
  conjunction: string
}

export interface TurkSessionInterface {
  id: number
  activity_id: number,
  expires_at: string,
  expired: boolean
}

export interface PassagesInterface {
  id?: number,
  text: string,
  image_link?: string,
  image_alt_text?: string,
  highlight_prompt?: string
}

export interface RuleInterface {
  id?: number,
  uid?: string,
  name: string,
  note?: string,
  universal: boolean,
  rule_type: string,
  optimal: boolean,
  suborder: number,
  concept_uid: string,
  prompt_ids?: number[],
  state: string,
  label?: {
    id: number,
    name: string
  },
  plagiarism_text?: {
    id?: number,
    rule_id?: number,
    text: string
  }
  regex_rules?: RegexRuleInterface[]
  regex_rules_attributes?: RegexRuleInterface[]
  feedbacks?: {
    id?: number,
    rule_id?: number,
    text: string,
    description?: string,
    order: number,
    highlights?: any[]
    highlights_attributes?: any[]
  }[]
}

export interface RuleFeedbackInterface {
  description?: string,
  highlights?: any[],
  highlights_attributes?: any[],
  id?: number,
  order?: number,
  rule_id?: number,
  text: string
}

export interface ActivitySessionInterface {
  activity_id: number,
  because_attempts: number,
  but_attempts: number,
  complete: boolean,
  flags: string[],
  scored_count: number,
  session_uid: string,
  so_attempts: number,
  start_date: string,
  strong_count: number,
  weak_count: number
}

export interface ActivitySessionsInterface {
  current_page: number,
  total_pages: number,
  activity_sessions: ActivitySessionInterface[]
}

export interface DropdownObjectInterface {
  value: string,
  label: string
}

export interface PromptHealthInterface {
  session_count: number,
  total_responses: number,
  display_name: number,
  num_final_attempt_optimal: number,
  num_final_attempt_not_optimal: number,
  avg_attempts_to_optimal: number,
  num_sessions_with_consecutive_repeated_rule: number,
  num_sessions_with_non_consecutive_repeated_rule: number,
  num_first_attempt_optimal: number,
  num_first_attempt_not_optimal: number
}

export type InputEvent = React.ChangeEvent<HTMLInputElement>;
export type ClickEvent = React.MouseEvent<HTMLElement>;
