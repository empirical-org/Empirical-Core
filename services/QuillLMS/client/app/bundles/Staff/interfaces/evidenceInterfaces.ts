export interface ActivityRouteProps {
  activityId: string,
  type?: string,
  promptConjunction?: string
}

export interface InvalidHighlight {
  rule_id: number,
  rule_type: string,
  prompt_id?: number
}

export interface ActivityInterface {
  id?: string,
  parent_activity_id?: string,
  flag?: string,
  title: string,
  notes: string,
  scored_level?: string,
  target_level?: number,
  version: number,
  passages?: PassagesInterface[],
  prompts?: PromptInterface[],
  passage_attributes?: PassagesInterface[],
  prompt_attributes?: PromptInterface[],
  invalid_highlights?: InvalidHighlight[]
}

export interface PromptInterface {
  id?: number,
  conjunction: string,
  text: string,
  max_attempts: number,
  max_attempts_feedback: string,
  first_strong_example: string,
  second_strong_example: string
}

export interface RegexRuleInterface {
  id?: number,
  rule_id?: number,
  regex_text: string,
  case_sensitive: boolean,
  sequence_type: string,
  conditional: boolean,
}

export interface FlagInterface {
  label: string,
  value: {}
}

export interface ActivityRuleSetPrompt {
  id?: number,
  conjunction: string
}

export interface TurkSessionInterface {
  id: number
  activity_id: number,
  expires_at: string,
  expired: boolean
}

export interface PassagesInterface {
  id?: number,
  text: string,
  image_link?: string,
  image_alt_text?: string,
  image_caption?: string,
  image_attribution?: string,
  highlight_prompt?: string,
  essential_knowledge_text?: string
}

export interface PlagiarismText {
  id?: number,
  rule_id?: number,
  text: string
}

export interface RuleInterface {
  id?: number,
  uid?: string,
  name: string,
  note?: string,
  universal: boolean,
  rule_type: string,
  optimal: boolean,
  suborder: number,
  concept_uid: string,
  prompt_ids?: number[],
  state: string,
  conditional?: boolean,
  hint?: {
    id: number,
    image_link: string,
    explanation: string
  },
  label?: {
    id: number,
    name: string
  },
  plagiarism_texts?: PlagiarismText[],
  regex_rules?: RegexRuleInterface[]
  regex_rules_attributes?: RegexRuleInterface[]
  feedbacks?: {
    id?: number,
    rule_id?: number,
    text: string,
    description?: string,
    order: number,
    highlights?: any[]
    highlights_attributes?: any[]
  }[]
}

export interface RuleFeedbackInterface {
  description?: string,
  highlights?: any[],
  highlights_attributes?: any[],
  id?: number,
  order?: number,
  rule_id?: number,
  text: string
}

export interface ActivitySessionInterface {
  activity_id: number,
  because_attempts: number,
  but_attempts: number,
  complete: boolean,
  flags: string[],
  scored_count: number,
  session_uid: string,
  so_attempts: number,
  start_date: string,
  strong_count: number,
  weak_count: number
}

export interface ActivitySessionsInterface {
  current_page: number,
  total_pages: number,
  activity_sessions: ActivitySessionInterface[]
}

export interface DropdownObjectInterface {
  value: any,
  label: string
}

export interface PromptHealthInterface {
  session_count: number,
  total_responses: number,
  display_name: number,
  num_final_attempt_optimal: number,
  num_final_attempt_not_optimal: number,
  avg_attempts: number,
  num_sessions_with_consecutive_repeated_rule: number,
  num_sessions_with_non_consecutive_repeated_rule: number,
  num_first_attempt_optimal: number,
  num_first_attempt_not_optimal: number,
  avg_time_spent: number,
  avg_confidence: number
}

export type InputEvent = React.ChangeEvent<HTMLInputElement>;
export type TextAreaEvent = React.ChangeEvent<HTMLTextAreaElement>;
export type ClickEvent = React.MouseEvent<HTMLElement>;
