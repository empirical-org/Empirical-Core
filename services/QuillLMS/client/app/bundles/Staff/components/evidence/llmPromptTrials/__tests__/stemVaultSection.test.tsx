import * as React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import StemVaultSection from '../stemVaultSection';
import { StemVaultInterface } from '../../../../interfaces/evidenceInterfaces';

jest.mock('../newDatasetModal', () => jest.fn(({ closeModal }) => (
  <div>
    <button onClick={closeModal}>Close Modal</button>
    <div>NewDatasetModal</div>
  </div>
)));

const mockStemVault: StemVaultInterface = {
  activity_id: 1,
  conjunction: 'because',
  created_at: '2023-01-01T12:00:00Z',
  datasets: [
    {
      id: 1,
      optimal_count: 10,
      suboptimal_count: 5,
      version: 1,
      created_at: '2023-01-01T12:00:00Z',
      notes: 'Sample dataset notes',
      trial_count: 3,
    },
  ],
  id: 1,
  prompt_id: 1,
  stem: 'Sample Stem',
  updated_at: '2023-01-02T12:00:00Z',
};

const renderComponent = (stemVault: StemVaultInterface) => {
  return render(
    <StemVaultSection stemVault={stemVault} />
  );
};

describe('StemVaultSection', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders the section with the correct title and dataset table', () => {
    renderComponent(mockStemVault);

    expect(screen.getByText('Because Datasets')).toBeInTheDocument();

    expect(screen.getByText('Dataset 1')).toBeInTheDocument();
    expect(screen.getByText('Sample dataset notes')).toBeInTheDocument();
    expect(screen.getByText('15')).toBeInTheDocument(); // Total test responses (10 + 5)
    expect(screen.getByText('10')).toBeInTheDocument(); // Optimal test responses
    expect(screen.getByText('5')).toBeInTheDocument(); // Suboptimal test responses
    expect(screen.getByText('3')).toBeInTheDocument(); // Trials
  });

  it('opens the NewDatasetModal when the New button is clicked', () => {
    renderComponent(mockStemVault);

    fireEvent.click(screen.getByText('New'));

    expect(screen.getByText('NewDatasetModal')).toBeInTheDocument();
  });

  it('closes the NewDatasetModal when the close button in the modal is clicked', () => {
    renderComponent(mockStemVault);

    fireEvent.click(screen.getByText('New'));

    fireEvent.click(screen.getByText('Close Modal'));

    expect(screen.queryByText('NewDatasetModal')).not.toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = renderComponent(mockStemVault);

    expect(asFragment()).toMatchSnapshot();
  });
});
