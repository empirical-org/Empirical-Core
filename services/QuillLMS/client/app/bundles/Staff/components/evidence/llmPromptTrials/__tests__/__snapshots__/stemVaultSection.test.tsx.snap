// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StemVaultSection matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="stem-vault-section"
  >
    <h5>
      <span>
        Because Datasets
      </span>
      <button
        class="quill-button small outlined"
        type="button"
      >
        New
      </button>
    </h5>
    <table
      class="data-table dataset-table"
    >
      <thead>
        <tr
          class="data-table-headers"
        >
          <th
            class="data-table-header undefined"
            scope="col"
            style="width: 70px; min-width: 70px; text-align: left;"
          >
            <span>
              Dataset
            </span>
          </th>
          <th
            class="data-table-header undefined"
            scope="col"
            style="width: 70px; min-width: 70px; text-align: left;"
          >
            <span>
              Created
            </span>
          </th>
          <th
            class="data-table-header undefined"
            scope="col"
            style="width: 400px; min-width: 400px; text-align: left;"
          >
            <span>
              Notes
            </span>
          </th>
          <th
            class="data-table-header center-content"
            scope="col"
            style="width: 54px; min-width: 54px; text-align: right;"
          >
            <span>
              Total Test Responses
            </span>
          </th>
          <th
            class="data-table-header center-content"
            scope="col"
            style="width: 71px; min-width: 71px; text-align: right;"
          >
            <span>
              Optimal Test Responses
            </span>
          </th>
          <th
            class="data-table-header center-content"
            scope="col"
            style="width: 85px; min-width: 85px; text-align: right;"
          >
            <span>
              Suboptimal Test Responses
            </span>
          </th>
          <th
            class="data-table-header center-content"
            scope="col"
            style="width: 30px; min-width: 30px; text-align: right;"
          >
            <span>
              Trials
            </span>
          </th>
          <th
            class="data-table-header center-content"
            scope="col"
            style="width: 70px; min-width: 70px; text-align: left;"
          >
            <span>
              Access
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        class="data-table-body"
      >
        <tr
          class="data-table-row  "
        >
          <td
            class="data-table-row-section undefined"
            style="width: 70px; min-width: 70px; text-align: left;"
          >
            Dataset 1
          </td>
          <td
            class="data-table-row-section undefined"
            style="width: 70px; min-width: 70px; text-align: left;"
          >
            12/31/22
          </td>
          <td
            class="data-table-row-section allow-wrap"
            style="width: 400px; min-width: 400px; text-align: left;"
          >
            Sample dataset notes
          </td>
          <td
            class="data-table-row-section center-content"
            style="width: 54px; min-width: 54px; text-align: right;"
          >
            15
          </td>
          <td
            class="data-table-row-section center-content"
            style="width: 71px; min-width: 71px; text-align: right;"
          >
            10
          </td>
          <td
            class="data-table-row-section center-content"
            style="width: 85px; min-width: 85px; text-align: right;"
          >
            5
          </td>
          <td
            class="data-table-row-section center-content"
            style="width: 30px; min-width: 30px; text-align: right;"
          >
            3
          </td>
          <td
            class="data-table-row-section undefined"
            style="width: 70px; min-width: 70px; text-align: left;"
          >
            <a
              class="quill-button extra-small outlined"
              href="/"
            >
              View
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </section>
</DocumentFragment>
`;
